framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,324 @@
+package org.checkerframework.common.value;
+
+import java.math.BigInteger;
+
+/**
+ * The Range Class with mathematics operations. Models the range indicated by the @IntRange
```

Comment:
The Range class seems very useful.  Could you please write some test cases for it, to help verify its correctness and to guard against regressions if anyone edits it in the future?